.github/workflows/build.yml

```diff
@@ -305,7 +305,13 @@ jobs:
         env:
           IMAGE_TAG: ${{ steps.tag.outputs.image_tag }}
           PROJECT_NAME: ${{ steps.setup.outputs.project_name }}
-        run: echo "name=${_AZ_REGISTRY}/${PROJECT_NAME}:${IMAGE_TAG}" >> $GITHUB_OUTPUT
+          SHA: ${{ github.sha }}
+        run: |
+          NAME="${_AZ_REGISTRY}/${PROJECT_NAME}:${IMAGE_TAG}"
+          if [[ "${IMAGE_TAG}" == "dev" ]]; then
+            NAME=$NAME",${_AZ_REGISTRY}/${PROJECT_NAME}:${SHA}"
+          fi
+          echo "name=$NAME" >> $GITHUB_OUTPUT
```

Comment:
If we're going to add multiple tags I think we should update the name of the steps related to be more accurate. Instead of `Generate image full name` we should have something like `Generate tag(s)`.  Also, do we want the full `SHA` by itself as a tag, or something like `dev-a1b2c3`?

Reply:
I think `dev` with short commit would be more readable, but from the card I assumed it should be a full commit hash.

Reply:
Let's use the full hash, but still include the tag name so it's easily identifiable. `${_AZ_REGISTRY}/${PROJECT_NAME}:dev-${SHA}`

Reply:
I didn't specify the commit has in the planned work, because I wanted it to be up for discussion. @michalchecinski makes a good point, a short commit would be easier to read, however, GitHub provides the `GITHUB_SHA` as a full commit. There will need to be assumptions and logic to shorten that when looking/pulling for an image as well.

Reply:
I made it short with `dev` prefix. Let me know if that's okay.